lib/spack/spack/package_prefs.py

```diff
@@ -15,6 +17,8 @@
 _lesser_spec_types = {'compiler': spack.spec.CompilerSpec,
                       'version': VersionList}
 
+_show_sgid_warning = True
```

Comment:
Can we avoid using a global by setting options to `warnings.warn`?

Reply:
That seems to work (thanks for the hint, I didn't know about `warnings.warn`). We now get one warning per package (better than before) but also a second warning per package from the forked build environment (not so nice but acceptable, I guess).